sdk-core/src/main/java/care/data4life/sdk/listener/Callback.kt

```diff
@@ -13,16 +13,14 @@
  * applications and/or if you’d like to contribute to the development of the SDK, please
  * contact D4L by email to help@data4life.care.
  */
+package care.data4life.sdk.listener
 
-package care.data4life.sdk.listener;
-
-import care.data4life.sdk.lang.D4LException;
+import care.data4life.sdk.lang.D4LException
 
 /**
  * Callback is a listener used when there is only success or error.
  */
-public interface Callback {
-    void onSuccess();
-
-    void onError(D4LException exception);
+interface Callback {
+    fun onSuccess()
+    fun onError(exception: D4LException)
```

Comment:
Do we want to make D4LExceptions generic?

Reply:
We have to do some exception hiding in case there is an error in a component that could leak keys. So we wrap every error in a D4LException.